store/migration/dev/20221126000000##principal_type.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE principal DROP CONSTRAINT principal_type_check;
+ALTER TABLE principal ADD CONSTRAINT principal_type_check CHECK (type IN ('END_USER', 'SYSTEM_BOT', 'SERVICE_ACCOUNT'));
```

Comment:
I am thinking to just use SYSTEM_BOT + @service.bytebase.com as the service account.

Otherwise, it's hard to tell the difference between SYSTEM_BOT and SERVICE_ACCOUNT

Reply:
Similar to borg service account v.s. IAM service account (but different), it might be good to separate the space between systembot v.s. service account.

Reply:
I think the `SYSTEM_BOT` with a specific email suffix is unclear and hard to maintain in the future. We need to hardcode the email suffix, and distinguish between the "service account" with our "bot account" (only the service account needs to use the token instead of the password or password_hash.

Also, for frontend, we need more work (also need the hardcode) to create this type of principal, seems users need to provide the bot name instead of the email, so we cannot reuse the member creation UX

Reply:
Or putting it another way. We have 2 use cases, one is the system bot we have now, the other is the service account. If we design it from scratch, would we create 2 separate types or just call it service account?

I am worried it's subtle for future readers to understand the difference.

Reply:
To clarify, I intend to provide more info. Having a separate type is also fine to me.

Reply:
Service accounts are used by Bytebase users, and system bots (system service accounts) are used by Bytebase internal (us). It will be great if we have another type to distinguish them if they are both service accounts, but a third type seems to be easier.